tests/test_filter.py

```diff
@@ -0,0 +1,41 @@
+"""Test the screening functionality."""
+
+from pathlib import Path
+from .utils_test import PATH_TEST
+from swan.filter.screen import main, apply_filters, read_molecules
+from swan.utils import Options
+import argparse
+import os
+
+path_input_test_filter = PATH_TEST / "input_test_filter.yml"
+
+
+def test_filter_cli(mocker) -> None:
+    """Test that the CLI works correctly."""
+    mocker.patch("argparse.ArgumentParser.parse_args", return_value=argparse.Namespace(
+        i=path_input_test_filter))
+
+    mocker.patch("swan.filter.screen.apply_filters", return_value=None)
+    main()
+
+
+def test_filter_functional_groups(tmp_path) -> None:
```

Comment:
The `@nanoutils.delete_finally()` decorator might be usefull here ([docs](https://nano-utils.readthedocs.io/en/latest/7_testing_utils.html#nanoutils.delete_finally)) if you want to get rid of the `try` / `finally` clause.

Note that both `nanoutils` and `assertionlib` (see comment below) are already required by `CAT`, so this shouldn't add any additional dependencies.
```suggestion
@delete_finally(f"path/to/my/candidates.csv")
def test_filter_functional_groups(tmp_path) -> None:
```

Reply:
Thanks for the suggestion. I have two issues with the decorator:
1. I cannot give a static string with the path to `candidates.csv` because it is generated inside a `tmp_path` that pytest generates at runtime.
2. I would prefer to copy-paste the decorator instead of using the dependency of a dependency because that can cause issues in the future.

Reply:
Yeah, fair enough.
Out of curiosity, how exactly does this `tmp_path` parameter work in combination with `pytest`?

Reply:
When you invoke pytest and the test function has the `tmp_path` argument, pytest automatically generates a temporal folder for you using (a platform specific) temporal folder. e.g. /tmp in Linux. Basically, you don't need to do anything to generate the temporal folder, you just assume that pytest will generate it at runtime.